src/noc/containers/noc-home/noc-home.component.ts

```diff
@@ -0,0 +1,134 @@
+
+  public caseRefVisibilityStates = caseRefVisibilityStates;
+  public caseRefErrorStates = [NocState.CASE_REF_SUBMISSION_FAILURE];
+
+  public qAndAVisibilityStates = qAndAVisibilityStates;
+  public answerErrorVisibilityStates = answerErrorVisibilityStates;
+
+  public checkAnswerVisibilityStates = checkAnswerVisibilityStates;
+  public nocSubmitSuccessStates = nocSubmitSuccessStates;
+
+  constructor(
+    private readonly store: Store<fromFeature.State>,
+    private readonly router: Router
+  ) { }
+
+  public ngOnInit() {
+    this.nocNavigationCurrentStateSub = this.store.pipe(select(fromFeature.currentNavigation)).subscribe(state => this.nocNavigationCurrentState = state);
+  }
+
+  public onNavEvent(event: NocNavigationEvent) {
+    this.navEvent = {
+      event,
+      timestamp: Date.now()
+    };
+    this.navigationHandler(event);
+  }
+
+  public isComponentVisible(currentNavigationState: NocState, requiredNavigationState: NocState[]): boolean {
+    return requiredNavigationState.includes(currentNavigationState);
+  }
+
+  public navigationHandler(navEvent: NocNavigationEvent) {
+    switch (navEvent) {
+      case NocNavigationEvent.BACK: {
+        switch (this.nocNavigationCurrentState) {
+          case NocState.START:
+            this.router.navigateByUrl('').then(r => {
+              return;
+            });
+            break;
+          case NocState.QUESTION:
+          case NocState.ANSWER_INCOMPLETE:
+          case NocState.ANSWER_SUBMISSION_FAILURE:
+            this.store.dispatch(new fromFeature.Reset());
+            break;
+          case NocState.CHECK_ANSWERS:
+            this.store.dispatch(new fromFeature.ChangeNavigation(NocState.QUESTION));
+            break;
+          case NocState.SUBMISSION_SUCCESS_PENDING:
+          case NocState.SUBMISSION_SUCCESS_APPROVED:
+          case NocState.SUBMISSION_FAILURE:
+            this.store.dispatch(new fromFeature.Reset());
+            break;
+          default:
+            throw new Error('Invalid NoC state');
+        }
+        break;
+      }
+      case NocNavigationEvent.CONTINUE: {
+        switch (this.nocNavigationCurrentState) {
+          case NocState.START:
+            this.nocCaseRefComponent.navigationHandler(navEvent);
+            break;
+          case NocState.CASE_REF_VALIDATION_FAILURE:
+            this.nocCaseRefComponent.navigationHandler(navEvent);
+            break;
+          default:
+            throw new Error('Invalid NoC state');
+        }
+        break;
+      }
+      case NocNavigationEvent.SET_ANSWERS: {
+        switch (this.nocNavigationCurrentState) {
+          case NocState.QUESTION:
+          case NocState.ANSWER_INCOMPLETE:
+            this.nocQandAComponent.navigationHandler(navEvent);
+            break;
+          default:
+            throw new Error('Invalid NoC state');
+        }
+        break;
+      }
+      case NocNavigationEvent.CHECK_ANSWERS: {
+        switch (this.nocNavigationCurrentState) {
```

Comment:
Given this switch only has one case, it could easily be an if. :)

Reply:
I will keep this as if change to if the code looks ood and not consistent.